### PR TITLE
Presales ZD Chat: use loadScript

### PR DESCRIPTION
**Pull request by yoavf**
The ZD presales chat widget was merged in #70594 while I was still working on some review feedback received in both #70118 and #70213. This implements the feedback into the component.

#### Proposed Changes
* Use `loadScript` to load the script, following the changes done in #70459 (+ updated tests)
* Rename `chatId` to `chatKey` for consistency

Note: I initially added an error boundary to the component (see [4f8d204](https://github.com/Automattic/wp-calypso/commit/4f8d204e4e80bf5c05dab44a4164208b3887970a)), as [suggested](https://github.com/Automattic/wp-calypso/pull/70213#discussion_r1033093875) by @niranjan-uma-shankar but I believe this is no longer relevant when using `loadScript`, since we're adding to script to `head` (happy to be corrected if an error boundary still makes sense)

#### Testing Instructions

* The widget is currently active on the JP Cloud pricing page - see #70590 for testing instructions

#### Pre-merge Checklist

- [x] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

Related to #

### CODE REVIEW

File: client/components/presales-zendesk-chat/index.tsx [hunk 1]
```diff
@@ -1,27 +1,26 @@
...
 
-		const script = document.createElement( 'script' );
-		script.src = 'https://static.zdassets.com/ekr/snippet.js?key=' + encodeURIComponent( chatId );
-		script.type = 'text/javascript';
-		script.id = 'ze-snippet';
+		loadScript(
+			'https://static.zdassets.com/ekr/snippet.js?key=' + encodeURIComponent( chatKey ),
+			() => {
+				return null;
+			},
```
Comment (tyxla) [inline client/components/presales-zendesk-chat/index.tsx]: Why is that necessary? What would be the downside of `loadScript()` returning a promise in this case?

Comment (yoavf) [reply client/components/presales-zendesk-chat/index.tsx]: > Why is that necessary? What would be the downside of `loadScript()` returning a promise in this case?

I might be wrong, but to pass the third argument to `loadScript` (optional params), we have to pass the 2nd callback arg, for which the type is defined as a function. And regardless of which function we pass, `loadScript` will not return a promise. 

I think the best I can do here is handle the error if it's returned, but I'm not sure it's worth it.
What do you think?

Comment (tyxla) [reply client/components/presales-zendesk-chat/index.tsx]: Well, we could pass `undefined` as the second argument, no? In that case, it would return a promise but I don't see a problem with that TBH - let me know if I'm missing something.

And as for handling the error - ideally, we should make sure that if something goes wrong with the script, it doesn't take the entire app down. I believe we still have to wrap it in an error boundary - we do load the script to the `head`,  but if it fatals, it still should throw inside that same component tree. Would be worth testing it out just in case, of course. IIRC, last time we did something like this, an error boundary helped.

Comment (yoavf) [reply client/components/presales-zendesk-chat/index.tsx]: > Well, we could pass undefined as the second argument, no?

That would require updating `loadScript`/`loadjQueryDependentScript` type definitions to accept that, but it makes sense - I'll do that. 

> Would be worth testing it out just in case

I'll try it!

Comment (yoavf) [reply client/components/presales-zendesk-chat/index.tsx]: @tyxla So I tried to simulate an error in a script loaded via a script tag - but as far as I can tell, this won't directly affect the rest of the app. If we were then to interact with the JS we are loading (i.e, like we did with the Olark implementation), then it would make sense to add an error boundary. 

But in this case, we don't - the Zendesk JS does it's own thing, and errors within the loaded script shouldn't affect Calypso.

Comment (tyxla) [reply client/components/presales-zendesk-chat/index.tsx]: What you're saying makes sense to me 👍 As long as we've weighed the risk and made sure that there is none, we should be good.

**Comment by yoavf** [comment]
Confirming I've tested this with the active ZD widget on the JP cloud pricing page, and it works as expected.

**Review by tyxla — COMMENTED** [review]
The code looks great from my perspective 👍 🚀 

I prefer leaving the final testing and approval to someone familiar with the new feature.

**Review by mattgawarecki — APPROVED** [review]
I can confirm the chat component still shows up (or is absent) when it should, and I like the improvements you've made @yoavf. I'm confident this is ready to ship :shipit:
